src/components/ProductItem.jsx

```diff
@@ -102,11 +135,16 @@ const ProductItem = ({ book:
                 </div>            
             </div>
             <div className="card-footer mt-1 mb-0-75  px-0-75">
-                <button className="btn btn-primary btn-text-icon btn-full-width p-0-25">
-                    Add to cart
+                <button className="btn btn-primary btn-text-icon btn-full-width p-0-25" 
+                    onClick={e => isAuth ? bookInCart ? navigate('/cart') : handleAddToCart()
```

Comment:
the argument e we aren't using, so can we keep it as just onClick={()=>your logic }

Reply:
Yes, sure. I have made the change in a different branch for this.